tests/nlp/checks/model_evaluation/single_dataset_performance_nlp_test.py

```diff
@@ -108,6 +108,41 @@ def test_run_with_scorer_multilabel_class_names(text_multilabel_classification_d
     assert_that(result.value.values[0][0], equal_to('a'))
 
 
+def test_run_with_scorer_multilabel_class_names_new_classes_in_observed(text_multilabel_classification_dataset_mock):
+    # Arrange
+    text_multilabel_classification_dataset_mock_copy = text_multilabel_classification_dataset_mock.copy()
+    text_multilabel_classification_dataset_mock_copy._classes = ['a', 'b', 'c']
+    text_multilabel_classification_dataset_mock_copy._label = [[0, 0, 1], [0, 1, 0], [0, 1, 0]]
+    check = SingleDatasetPerformance(scorers=['f1_per_class'])
+
+    # Act
+    result = check.run(text_multilabel_classification_dataset_mock_copy,
+                       predictions=[[0, 0, 1], [1, 0, 1], [0, 1, 0]],
+                       model_classes=[[0, 1], [0, 1], [0, 1]]
```

Comment:
As said in the comments above, IMO model_classes should be the one conveying the info that the class names are `['a', 'b', 'c']`